### PR TITLE
Set up tracing via Open Telemetry

### PR DESCRIPTION
**Pull request by iancanderson**
And publish spans for overall request and each fragment fetch

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -43,10 +43,33 @@ server.ListenAndServe()
...
+- `viewproxy` is not coupled to a specific application framework, but _is_ being driven by close integration with Rails applications.
 - `viewproxy` should rely on Rails' (or other target application framework) strengths when possible.
 - `viewproxy` itself and its client API's should focus on developer happiness and productivity.
+
+## Development
+
+Run the tests:
+
+```
+go test
```
Comment (BlakeWilliams) [inline README.md:74]:
```suggestion
go test ./...
```

Should this test all packages?

File: pkg/tracing/tracing.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
+package tracing
```
Comment (BlakeWilliams) [inline pkg/tracing/tracing.go:1]: Should this be moved to the `internal` directory?

File: server.go [hunk 3]
```diff
@@ -44,6 +47,8 @@ type Server struct {
 	HmacSecret string
 	// A function that is called before the request is handled by viewproxy.
 	PreRequest func(w http.ResponseWriter, r *http.Request)
+  tracingConfig tracing.TracingConfig
```
Comment (BlakeWilliams) [inline server.go:50]: Indentation looks a bit off, can we run `go fmt`?

File: server.go [hunk 4]
```diff
@@ -126,14 +132,28 @@ func (s *Server) matchingRoute(path string) (*Route, map[string]string) {
 }
 
 func (s *Server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	ctx := context.Background()
+
+	shutdownTracing, err := tracing.Instrument(s.tracingConfig, s.Logger)
```
Comment (BlakeWilliams) [inline server.go:137]: `http.ListenAndServe` should be blocking, could this be moved to our `ListenAndServe` function so we create only a single tracing instance?

**Review by BlakeWilliams — APPROVED** [review]
Added a few comments, nothing blocking though! 👍

File: server.go (hunk 3 above)
Comment (iancanderson) [reply server.go]: Fixed in db650657d00931e3531d8225979a1489d37034ad

File: server.go (hunk 4 above)
Comment (iancanderson) [reply server.go:136]: oooooh I couldn't wrap my head around that before but that makes sense 👍

Comment (iancanderson) [reply server.go]: done in ad39aaab08c4679858074f592e61bdaa69573405

File: pkg/tracing/tracing.go (hunk 2 above)
Comment (iancanderson) [reply pkg/tracing/tracing.go]: Done in d3a51f9e6e8a91eccf6c1dfb50b5ca0fba7a2f9e

**Pull request merged**